uploadNeuroDB/imaging_non_minc_insertion.pl

```diff
@@ -0,0 +1,583 @@
+}
+unless ( $output_type ) {
+    print STDERR "$Usage\n\tERROR: missing -output_type argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+unless ( $scan_type ) {
+    print STDERR "$Usage\n\tERROR: missing -scan_type argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+unless ( $date_acquired ) {
+    print STDERR "$Usage\n\tERROR: missing -date_acquired argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+unless ( $scanner_id ) {
+    print STDERR "$Usage\n\tERROR: missing -scanner_ID argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+unless ( $coordin_space ) {
+    print STDERR "$Usage\n\tERROR: missing -coordin_space argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+unless ( $upload_id ) {
+    print STDERR "$Usage\n\tERROR: missing -upload_id argument\n\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+
+# Make sure the files specified as an argument exist and are readable
+unless (-r $file_path) {
+    print STDERR "$Usage\n\tERROR: You must specify a valid file path to "
+                 . "insert using the -file_path option.\n\n";
+    exit $NeuroDB::ExitCodes::INVALID_PATH;
+}
+# Make sure that the metadata file is readable if it is set
+if ( $metadata_file && !(-r $metadata_file) ){
+    print STDERR "\n\tERROR: The metadata file does not exist in the filesystem.\n\n";
+    exit $NeuroDB::ExitCodes::INVALID_PATH;
+}
+
+
+
+
+###### Establish database connection
+
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+
+
+
+###### Get config settings
+
+my $data_dir = NeuroDB::DBI::getConfigSetting(\$dbh, 'dataDirBasepath');
+
+
+
+
+###### For the log, temp directories and notification spools
+
+# determine local time
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time);
+my $today = sprintf( "%4d-%02d-%02d %02d:%02d:%02d",
+                     $year+1900,$mon+1,$mday,$hour,$min,$sec
+);
+
+# create the temp directory
+my $template = "FileLoad-$hour-$min-XXXXXX";
+my $TmpDir   = tempdir($template, TMPDIR => 1, CLEANUP => 1 );
+my @temp     = split(/\//, $TmpDir);
+my $temp_log = $temp[$#temp];
+
+# create the log file
+my $log_dir  = $data_dir . "/logs";
+my $log_file = $log_dir . "/" . $temp_log . ".log";
+my $message  = "\nlog dir is $log_dir and log file is $log_file.\n";
+print $message if $verbose;
+
+# open log file and write successful connection to DB
+open( LOG, ">>", $log_file ) or die "\nError Opening $log_file.\n";
+LOG->autoflush(1);
+&logHeader();
+$message = "\n==> Successfully connected to database\n";
+print LOG $message;
+
+# create Notify and Utility objects
+my $notifier = NeuroDB::Notify->new(\$dbh);
+my $utility = NeuroDB::MRIProcessingUtility->new(
+    \$dbh, 0, $TmpDir, $log_file, $verbose
+);
+
+
+
+
+##### Verify that the provided scanner ID refers to a valid scanner entry
+my $query = "SELECT ID FROM mri_scanner WHERE ID=?";
+my $sth = $dbh->prepare($query);
+$sth->execute($scanner_id);
+unless ($sth->rows > 0) {
```

Comment:
I'd reuse method `getScannerCandID` from `MRI.pm` here, like so

```perl
unless(defined NeuroDB::MRI::getScannerCandID($scanner_id, \$dbh)) {
```

This eliminates the need for `$query` and `$sth`.

Reply:
Good idea. Done in the next commit. Thanks!